README.md

```diff
@@ -23,24 +38,137 @@
+`sensuctl asset add sensu-plugins/sensu-plugins-memory-checks`
 
-[Installation and Setup](http://sensu-plugins.io/docs/installation_instructions.html)
+If you're using an earlier version of sensuctl, you can download the asset definition from [this project's Bonsai Asset Index page](https://bonsai.sensu.io/assets/sensu-plugins/sensu-plugins-memory-checks).
+
+#### Asset definition
+
+```yaml
+---
+type: Asset
+api_version: core/v2
+metadata:
+  name: sensu-plugins-memory-checks
+spec:
+  url: https://assets.bonsai.sensu.io/c5391d4ae186484226732344b35cf95c0b07b8ec/sensu-plugins-memory-checks_4.0.0_centos_linux_amd64.tar.gz
+  sha512: ea297a85aa3612da7f78d948f9784443fffac511040c5130a2dcde7191a0004044c2ef881e665520cbc64431955ab19920d84de6b5fed85c63da7091c4b93bf0
+```
+
+#### Check definition
+
+```yaml
+---
+type: CheckConfig
+spec:
+  command: "check-memory.rb"
+  handlers: []
+  high_flap_threshold: 0
+  interval: 10
+  low_flap_threshold: 0
+  publish: true
+  runtime_assets:
+  - sensu-plugins/sensu-plugins-memory-checks
+  - sensu/sensu-ruby-runtime
+  subscriptions:
+  - linux
+```
+
+### Sensu Core
+
+#### Check definition
+```json
+{
+  "checks": {
+    "metrics-disk-usage": {
+      "command": "metric-disk-usage.rb",
+      "subscribers": ["linux"],
+      "interval": 10,
+      "refresh": 10,
+      "handlers": ["influxdb"]
+    }
+  }
+}
+```
+
+## Installation from source
+
+### Sensu Go
+
+See the instructions above for [asset registration](#asset-registration)
+
+### Sensu Core
+
+Install and setup plugins on [Sensu Core](https://docs.sensu.io/sensu-core/latest/installation/installing-plugins/)
+
+## Additional notes
 
 ### `bin/check-ram.rb`
 
-Ruby does not have a good native way without the use of C extensions to grab information on memory usage (outside of running shell commands and parsing out). Because of this `check-ram.rb` uses a gem called `vmstat` which has a somewhat annoying dependency on a gcc to compile the C extensions. In order to not force people to install gcc if they are not using `check-ram.rb` we do not install it by default, and is up to the user to make sure that they install it. This requires two steps and can vary based on your distribution and ruby set up:
-  * Install `gcc`. If you are on a debian based system it is located in the `build-essential` package.
-  * Install `vmstat` gem into the path that sensu gems are expected to run from:
-    * `sudo /opt/sensu/embedded/bin/gem install vmstat --no-rdoc --no-ri` or the equivalent for installing sensu gems in your configuration management system
+Ruby does not have a good native way to grab information on memory usage (without using C extensions or running shell commands and parsing out). For this reason, `check-ram.rb` uses a gem called `vmstat` that has a dependency on a GCC to compile the C extensions.
+
+Some users may not need to use `check-ram.rb`. These users do not need gcc to use the rest of this plugin, so we do not install GCC by default.
```

Comment:
```suggestion
To avoid errors which occur when trying to compile the C extensions without GCC present, we do not install `vmstat` gem by default.
```